whylog/config/rule.py

```diff
@@ -19,10 +19,10 @@ def serialize(self):
             "constraints": self._constraints,
         }
 
-    def get_new_parsers(self, old_parsers):
+    def get_new_parsers(self, parser_name_generator):
         new_parsers = []
         for parser in itertools.chain([self._effect], self._causes):
```

Comment:
Used chain is elegant, I want iterate over parsers belongs to rule not create list